projects/plugins/jetpack/modules/masterbar.php

```diff
@@ -25,6 +25,7 @@
 
 if ( jetpack_is_atomic_site() ) {
 	new Inline_Help();
+	require_once __DIR__ . '/masterbar/wp-posts-list/bootstrap.php';
```

Comment:
We could consider that this `masterbar.php` file is already bootstraping all the features bundled in the masterbar module (soon to be renamed to just the WP.com module).

Do you think if would be better to have everything centralized in this file rather than having "mini-bundles"? I understand that we're trying to avoid a very large file if eventually we have a ton of classes to load, but having everything in one single file seems much simpler to me.

That way we would be consistent with how everything else is being loaded:

```php
require __DIR__ . '/masterbar/masterbar/class-masterbar.php';
require __DIR__ . '/masterbar/admin-color-schemes/class-admin-color-schemes.php';
require __DIR__ . '/masterbar/inline-help/class-inline-help.php';
require __DIR__ . '/masterbar/wp-posts-list/class-posts-list-page-notification.php';

new Masterbar();
new Admin_Color_Schemes();

if ( jetpack_is_atomic_site() ) {
  new Inline_Help();
  new Posts_List_Page_Notification();
}
```

This would need to move to the constructor/init_actions the logic that checks whether the current screen is eligible for this feature.

Reply:
> We could consider that this masterbar.php file is already bootstraping all the features bundled in the masterbar module (soon to be renamed to just the WP.com module).

Imho, I think we should have masterbar.php responsible only as a glue code for the module and for having environment checks like limiting features only on AT, for example. Since the module will be renamed to WP.com module, the scope will change and additional implementations will be in here. Therefore, the script will mostly likely increase in size. 

> Do you think if would be better to have everything centralized in this file rather than having "mini-bundles"? I understand that we're trying to avoid a very large file if eventually we have a ton of classes to load, but having everything in one single file seems much simpler to me.

I do see your point, but on the other side, I'm worried that handling the bootstrap in masterbar.php would make things a bit harder to maintain. For example, if we have a new feature small feature related to `wp-posts-list` "mini-bundle" (I really like the name, haha) that would want to deploy on WPCOM, we would still need to duplicate the code in masterbar.php from WPCOM SVN repo for every class/feature we introduce. 

> That way we would be consistent with how everything else is being loaded:

Yes, but we already using a load.php "bootstrap" file for Admin-menu.

> This would need to move to the constructor/init_actions the logic that checks whether the current screen is eligible for this feature.

Imho, I think these kind of checks shouldn't be in the class because it would make it state coupled to a global environment variable. 

Also, having a bootstrap file also has an advantage if we want to have the possibility to restrict/control a mini-bundle, since we could enable/disable an entire group of features (interconnected, maybe?) at the same time.